src/kOS/Execution/TransferManager.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using kOS.Safe;
+using kOS.Suffixed;
+
+namespace kOS.Execution
+{
+    public class TransferManager : IUpdateObserver
+    {
+        public enum TransferStatus        
+        {
+            Failed,
+            Finished,
+            Inactive,
+            Transferring
+        }
+
+        private readonly SharedObjects shared;
+        private readonly HashSet<ResourceTransferValue> transfers;
+
+        public TransferManager(SharedObjects shared)
+        {
+            this.shared = shared;
+            transfers = new HashSet<ResourceTransferValue>();
+            shared.UpdateHandler.AddFixedObserver(this);
+        }
+
+        public ResourceTransferValue CreateTransfer(PartResourceDefinition resourceInfo, object transferTo, object transferFrom, double amount)
+        {
+            var toReturn = new ResourceTransferValue(this, resourceInfo, transferTo, transferFrom, amount);
+            transfers.Add(toReturn);
+            return toReturn;
+        }
+
+        public ResourceTransferValue CreateTransfer(PartResourceDefinition resourceInfo, object transferTo, object transferFrom)
+        {
+            var toReturn = new ResourceTransferValue(this, resourceInfo, transferTo, transferFrom);
+            transfers.Add(toReturn);
+            return toReturn;
+        }
+
+        public void Dispose()
+        {
+            shared.UpdateHandler.RemoveFixedObserver(this);
```

Comment:
This is the transfer manager which is in shared and will only get disposed when shared does. His job is to hold the list of transfers. He could subscribe and unsub when he doesnt have anything. or I could just leave it updating and add this to the first line.

    if (!transfers.Any()) return;

That's a pretty cheap check.